### PR TITLE
disconnected flag

### PR DESCRIPTION
**Pull request by mxpaspa**

### CODE REVIEW

File: reporter/v2/pkg/reporter/reconcile_task.go [hunk 1]
```diff
@@ -101,6 +101,67 @@ func (r *ReconcileTask) run(ctx context.Context) error {
 			logger.Error(err, "error uploading files from data service", details...)
 			return err
 		}
+	} else if !r.CanRunUploadTask(ctx) {
```
Comment (mxpaspa) [inline reporter/v2/pkg/reporter/reconcile_task.go:104]: @zach-source is this the right approach to update the report statuses ?

File: reporter/v2/pkg/reporter/reconcile_task.go [hunk 2]
```diff
@@ -158,7 +219,7 @@ func (r *ReconcileTask) ReportTask(ctx context.Context, report *marketplacev1alp
 }
 
 func (r *ReconcileTask) CanRunUploadTask(ctx context.Context) bool {
```
Comment (zach-source) [inline reporter/v2/pkg/reporter/reconcile_task.go:221]: CanRunUploadTask should be true if you can do it. If you return just `IsDisconnected` then that would flip this. Try inversing the value.

File: reporter/v2/pkg/reporter/reconcile_task.go [hunk 3]
```diff
@@ -158,7 +219,7 @@ func (r *ReconcileTask) ReportTask(ctx context.Context, report *marketplacev1alp
...
-	return true
+	return r.Config.IsDisconnected
```
Comment (zach-source) [inline reporter/v2/pkg/reporter/reconcile_task.go:222]:
```suggestion
	return !r.Config.IsDisconnected
```

File: reporter/v2/pkg/reporter/reconcile_task.go (hunk 1 above)
Comment (zach-source) [reply reporter/v2/pkg/reporter/reconcile_task.go:104]:
```suggestion
	} else if r.CanRunUploadTask(ctx) {
```

File: reporter/v2/pkg/reporter/reconcile_task.go [hunk 4]
```diff
@@ -101,6 +101,67 @@ func (r *ReconcileTask) run(ctx context.Context) error {
...
+			condition := marketplacev1alpha1.ReportConditionJobIsDisconnected
+			details := &marketplacev1alpha1.UploadDetails{}
+			details.Status = marketplacev1alpha1.UploadStatusNoop
+
+			if err := updateMeterReportStatus(ctx, r.K8SClient, metadata.ReportName, metadata.ReportNamespace,
+				func(m marketplacev1alpha1.MeterReportStatus) marketplacev1alpha1.MeterReportStatus {
+					m.Conditions.SetCondition(condition)
+
+					statuses = append(statuses, details)
+					m.UploadStatus.Append(statuses)
```
Comment (zach-source) [inline reporter/v2/pkg/reporter/reconcile_task.go:150]: Let's not update the UploadStatus. No need to put a row there. Conditions is fine.

File: reporter/v2/pkg/reporter/reconcile_task.go [hunk 5]
```diff
@@ -101,6 +101,67 @@ func (r *ReconcileTask) run(ctx context.Context) error {
...
+
+					dataServiceStatus := m.UploadStatus.Get(uploaders.UploaderTargetDataService.Name())
+
+					if dataServiceStatus != nil {
+						m.DataServiceStatus = dataServiceStatus
+					}
+
+					return m
+				}); err != nil {
+				logger.Error(err, "failed to update meter report")
```
Comment (zach-source) [inline reporter/v2/pkg/reporter/reconcile_task.go:160]:
```suggestion
			condition := marketplacev1alpha1.ReportConditionJobIsDisconnected

			if err := updateMeterReportStatus(ctx, r.K8SClient, metadata.ReportName, metadata.ReportNamespace,
				func(m marketplacev1alpha1.MeterReportStatus) marketplacev1alpha1.MeterReportStatus {
					m.Conditions.SetCondition(condition)
					return m
				}); err != nil {
				logger.Error(err, "failed to update meter report")
```

**Review by zach-source — CHANGES_REQUESTED** [review]

**Review by zach-source — APPROVED** [review]
Few small issues that came out with testing. I've added a commit to fix them.

**Pull request merged**